docs/concepts/crds/snapshot/index.md

```diff
@@ -5,145 +5,221 @@ menu:
     identifier: snapshot-overview
     name: Snapshot
     parent: crds
-    weight: 50
+    weight: 55
 product_name: kubestash
 menu_name: docs_{{ .version }}
 section_menu_id: concepts
 ---
-> New to Stash? Please start [here](/docs/concepts/README.md).
+> New to KubeStash? Please start [here](/docs/concepts/README.md).
 
 # Snapshot
 
 ## What is Snapshot
 
-A `Snapshot` is a representation of backup snapshot in a Kubernetes native way. Stash uses an [Aggregated API Server](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/api-machinery/aggregated-api-servers.md) to provide `get` and `list` capabilities for snapshots from the backend.
+A `Snapshot` is a Kubernetes `CustomResourceDefinition`(CRD) which represents the state of a backup run to a particular `Repository`. Multiple components of the same target may be backed up in the same `Snapshot`. This is a namespaced CRD. It should be in the same namespace as the respective `Repository`. 
 
-This enables you to view some useful information such as `creationTimestamp`, `snapshot id`, `backed up path` etc of a snapshot. This also provides the capability to restore a specific snapshot.
+KubeStash operator is responsible for creating `Snapshot` custom resource. `Snapshot` is not supposed to be created/edited by the end user.
```

Comment:
A `Snapshot` is a Kubernetes `CustomResourceDefinition`(CRD) which represents the state of a backup run for one or more components of an application. For every `BackupSession`, it creates `Snapshot` CRs. If a `BackupSession` involves multiple repositories, a `Snapshot` is created  for each repository.